filesystem/src/main/java/org/frankframework/filesystem/LocalFileSystem.java

```diff
@@ -419,14 +424,34 @@ public Map<String, Object> getAdditionalFileProperties(Path file) throws FileSys
 		}
 	}
 
-	private String readAttributeValue(Object attributeValue) {
-		if (attributeValue instanceof byte[] bytes) {
-			return new String(bytes);
-		} else if (attributeValue instanceof ByteBuffer buffer) {
-			return new String(buffer.array());
+	private @Nonnull String readAttributeValue(@Nonnull Path file, @Nonnull String name, @Nonnull Object attributeValue) {
+		try {
+			if (attributeValue instanceof byte[] bytes) {
+				return bytesToString(bytes);
+			} else if (attributeValue instanceof ByteBuffer buffer) {
+				return bytesToString(buffer.array());
+			} else {
+				return attributeValue.toString();
+			}
+		} catch (Exception e) {
+			log.warn(() -> "Error parsing file attribute [%s] on file [%s]:".formatted(name, file), e);
+			return "<attribute not readable>";
+		}
+	}
+
+	@Nonnull
+	private static String bytesToString(@Nonnull byte[] bytes) {
+		CharsetDetector detector = new CharsetDetector();
+		detector.setText(bytes);
+		CharsetMatch match = detector.detect();
+		Charset charset;
+		if (match.getConfidence() >= MessageUtils.CHARSET_CONFIDENCE_LEVEL) {
+			charset = Charset.forName(match.getName());
 		} else {
-			return attributeValue.toString();
+			// If confidence was low, use system default charset
+			charset = Charset.defaultCharset();
 		}
+		return XmlEncodingUtils.replaceNonValidXmlCharacters(new String(bytes, charset), '?', false, true);
```

Comment:
Van string naar bytes snap ik dat de charset nodig is, maar van bytes naar string to niet?
Ik neem aan dat het de `Charset.defaultCharset()` is?

Reply:
We gebruiken de charset hier om van `byte[]` naar `String` te gaan.

Wanneer charset niet opgegeven wordt dan is het altijd de `Charset.defaultCharset()` inderdaad.

Reply:
PS: Als de charset-decoder iets tegenkomt wat hij niet snapt, dan kan dat ipv een verkeerd character ook een exception geven.
In ieder geval, lijkt me geen micro-optimalisatie. Afdekken van een edge-case, maar wel een die eerder wat performance kost dan een micro-optimalisatie. 😬